File: 01.JS-Front-End-JS-Syntax-Fundamentals-More-Exercise/spiceMustFlow.js
function gatherSpice(initialYield) {
    let yield = initialYield
    let spiceMined = 0
    let totalDays = 0

    while (yield >= 100) {
        spiceMined += yield - 26
        totalDays += 1
        yield -= 10
    }

    if (spiceMined >= 26){
        spiceMined -= 26
    } else {
        spiceMined -= spiceMined
    }
    
    console.log(totalDays)
    console.log(spiceMined)
}

gatherSpice(111)
gatherSpice(450)

File: 01.JS-Front-End-JS-Syntax-Fundamentals-More-Exercise/spice_must_flow.py
def gather_spice(initial_yield):
    yield_amount = initial_yield
    spice_mined = 0
    total_days = 0

    while yield_amount >= 100:
        spice_mined += yield_amount - 26
        total_days += 1
        yield_amount -= 10
    
    if spice_mined >= 26: 
        spice_mined -= 26
    else: 
        spice_mined -= spice_mined

    print(total_days)
    print(spice_mined)

gather_spice(111)
gather_spice(450)